onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -2679,6 +2679,140 @@ It's an extension of Gelu. It takes the sum of input A and bias input B as the i
         updateOutputElemType(ctx, 0, ONNX_NAMESPACE::TensorProto::BOOL);
       });
 
+  static const char* OptionalConstruct_ver1_doc = R"DOC(
+      Construct an optional type containing either an empty optional of a certain type specified by the attribute, "
+      "or an optional type containing the 'input' element."
+      )DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(OptionalConstruct)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(OptionalConstruct_ver1_doc)
+      .Input(0, "input", "The input element.", "T", OpSchema::Optional)
+      .Attr("type", "Type of the element in the optional output", AttributeProto::TYPE_PROTO, OPTIONAL_VALUE)
+      .Output(0, "output", "The optional output enclosing the input element.", "O")
+      .TypeConstraint(
+          "T",
+          {"tensor(float)",
+           "seq(tensor(float))"},
+          "Constrains input type to all tensor and sequence types.")
+      .TypeConstraint(
+          "O",
+          {"optional(tensor(float))", "optional(tensor(uint8))", "optional(tensor(uint16))", 
+           "optional(tensor(uint32))", "optional(tensor(uint64))", "optional(tensor(int8))", 
+           "optional(tensor(int16))", "optional(tensor(int32))", "optional(tensor(int64))", 
+           "optional(tensor(float16))", "optional(tensor(float))", "optional(tensor(double))", 
```

Comment:
We already had **optional(tensor(float))** at the beginning. Does this line should be **float -> bfloat16**?